tests/test_AllocateSizeMoreThanClusterSize.py

```diff
@@ -0,0 +1,332 @@
+"""
+Automates the following test
+
+OCS-269 - FT-OCP-Create-PV-AllocateSizeMoreThanClusterSize
+Verify a PVC creation by allocating more storage
+than what is available in Ceph
+"""
+
+import logging
+import yaml
+import os
+import pytest
+import ocs.defaults as defaults
+
+from ocs import ocp
+from utility import utils, templating
+from utility.utils import run_cmd
+from ocsci.testlib import tier1, ManageTest
+
+log = logging.getLogger(__name__)
+
+PVC_YAML = os.path.join(
+    "templates", "RBD_PersistentVolumeClaim.yaml"
+)
+POD_YAML = os.path.join(
+    "templates", "RBD_NginxPod.yaml"
+)
+SC_YAML = os.path.join(
+    "templates", "RBD_StorageClass.yaml"
+)
+
+
+TEMP_PVC_YAML_FILE = '/tmp/pvc_test.yaml'
+TEMP_POD_YAML_FILE = '/tmp/pod_test.yaml'
+TEMP_SC_YAML_FILE = '/tmp/sc_test.yaml'
+
+POD = ocp.OCP(
+    kind='Pod', namespace=defaults.TEST_NAMESPACE
+)
+PVC = ocp.OCP(
+    kind='PersistentVolumeClaim', namespace=defaults.TEST_NAMESPACE
+)
+PV = ocp.OCP(
+    kind='PersistentVolume', namespace=defaults.TEST_NAMESPACE
+)
+SC = ocp.OCP(
+    kind='StorageClass', namespace=defaults.TEST_NAMESPACE
+)
+NAMESPACE = ocp.OCP(
+    kind='Project', namespace=defaults.TEST_NAMESPACE
+)
+OCP = ocp.OCP(
+    kind='Pod', namespace=defaults.TEST_NAMESPACE
+)
+
+
+@pytest.fixture(scope='class')
+def test_fixture(request):
+    """
+    Over allocate a PVC above cluster's overall capacity
+    """
+    self = request.node.cls
+
+    def finalizer():
+        teardown(self)
+    request.addfinalizer(finalizer)
```

Comment:
I think that `request.addfinalizer(teardown)` is enough and there doesn't need to be another layer (`finalizer` method).

Reply:
@fbalak note that `teardown` is called with `self` argument, and so this layer needs to be there as long as passing reference to `self` into `finalizer` function is intended. You could make it shorter using lambda, but that's about it :) I don't like this kind of code organization either, but I don't think it's a good idea to block this PR because of this, esp. when lot of other people got similar code merged. I would rather draw a line at some point when we get an agreement on this, change this at once and then start to enforce it during reviews.

Reply:
Ah, ok.